fdbserver/OldTLogServer_4_6.actor.cpp

```diff
@@ -412,7 +414,9 @@ namespace oldTLog_4_6 {
 				// These are initialized differently on init() or recovery
 				recoveryCount(), stopped(false), initialized(false), queueCommittingVersion(0), newPersistentDataVersion(invalidVersion), recovery(Void())
 		{
-			startRole(Role::TRANSACTION_LOG,interf.id(), UID());
+			std::map<std::string, std::string> details;
+			details["TLogVersion"] = TLOG_VERSION;
+			startRole(Role::TRANSACTION_LOG, interf.id(), tLogData->workerID, details, "Restored");
```

Comment:
You could just s/details/{{"TLogVersion", TLogVersion::V2}}/ as it doesn't look like you need the map after this anyway.

Reply:
True, though maybe if we add a second field for the spill type it may be cleaner to not have it all inline in the `startRole` call.

Reply:
(I am going to piggy back this comment here) It will make me feel much better if you can change `startRole` to take a reference, instead of copying, of its last two arguments.

Reply:
@dongxinEric Done